### PR TITLE
Improve the mxd test 43 v2

### LINKED ISSUE #43: CI: improve the MXD test
### WHAT

Currently, during CI runs, we only test that both `bob` and `alice` are up and running, check out [this workflow job](https://github.com/eclipse-tractusx/tutorial-resources/blob/c140d461667d640cab27c27983283a5eddafea81/.github/workflows/run-terraform.yml#L70). That is not very thorough and we should improve it. 

At least the following aspects should be tested:
- create an asset, policy, contract def
- make a catalog request (bob -> alice, alice -> bob)
- [optional] perform contract negotiation
- [optional] perform data transfer

For that, we could use another postman collection and leverage the test feature, or even just plain `curl` commands.

## WHY
Testing should be improved, MXD does indeed serve as a testing ground of sorts.

### PR DESCRIPTION
**Pull request by ravinderkumarsap**
Please describe your PR: 
- What does this PR introduce?   it will solve the issue raise **https://github.com/eclipse-tractusx/tutorial-resources/issues/43**
- Does it fix a bug?  it will improve the testing performance.
- Does it add a new feature? 
- Is it enhancing documentation? yes

This cover below postman collection

**•	Create an asset BOB, 
•	Create an asset Alice
•	Create policy BOB,
•	Create policy Alice 
•	Create contract def BOB
•	Create contract def Alice
•	Make a catalog request Alice to BOB
•	Make a catalog request BOB to Alice**

**We have created reuseable workflow for test issue and completed below comments as suggested.**

Executing Postman collections before the cluster is created **(Completed)**
Create test assertions in Postman **(Completed)**
Create separate Test file for reusable workflow **(Completed)**
Modify your Terraform workflow YAML file to invoke the test workflow **(Completed)**
Test locally **(Completed)**
Test after code push **(Completed)**

**Sample Screen Shot of RUN**

image

**Documentation**

[Issue 43.docx]

(https://github.com/eclipse-tractusx/tutorial-resources/files/12768709/Issue.43.docx)

Closes #43

### CODE REVIEW

File: mxd/postman/MXD_Management_API.json [hunk 1]
```diff
@@ -0,0 +1,600 @@
+{
```
Comment (paullatzelsperger) [inline mxd/postman/MXD_Management_API.json:1]: `management_api_tests.json` would be a better name for this file.

There are two variables `MANAGEMENT_URL` and `PROVIDER_PROTOCOL_URL` that are undefined in the collection, so running this via `newman` won't work. You probably have those variables in your local postman installation, which it why it works - locally.

Again: have you tested this locally **with newman**, or in a CI run?

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]
- a composite action is not the same thing as a reusable workflow
- variables in the collection are undefined

**Comment by paullatzelsperger** [comment]
furthermore: please don't attach Word documents. This is cumbersome for reviewers, provides 0 additional value, and forces the use of non-OSS tools in an OSS project. 
There also is no need to provide a comprehensive list of tasks that you completed. just a quick summary of the PR.

**Comment by ravinderkumarsap** [comment]
Let me work in suggested points. I will update you @paullatzelsperger

File: .github/actions/mxd-test-action/action.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations
+#  under the License.
+#
+#  SPDX-License-Identifier: Apache-2.0
+#
+---
+name: "Run Tests"
```
[the remark this answers is not in the document]
Comment (hemantxpatel) [reply .github/actions/mxd-test-action/action.yml:20]: Tests need to be executed as a `job step` after the `terraform apply` step. Reusable workflow can't be executed as a job step but the complete job. So if we want to run the test in a separate job / reusable workflow, we need to define all other steps as well (kinda duplicate of `run-terraform.yml` as `run-tests.yml`).

[the remark this answers is not in the document]
Comment (hemantxpatel) [reply .github/actions/mxd-test-action/action.yml:20]: I tried to make sequential jobs, first one deploys the cluster, second one run the tests, third one destroys the cluster.
[Gist](https://gist.github.com/hemantxpatel/127ecb6bd53753e04dfce8b154b2be72)

Second and third job couldn't access the cluster as it runs in a separate runner.

File: mxd/postman/MXD_Management_API.json (hunk 1 above)
Comment (ravinderkumarsap) [reply mxd/postman/MXD_Management_API.json:1]: @paullatzelsperger  I've removed the unused variables, namely MANAGEMENT_URL and PROVIDER_PROTOCOL_URL, as they are not being utilized within the collections.

yes these are tested using newman CI.

image

**Comment by ravinderkumarsap** [comment]
@paullatzelsperger  I have updated the request changes. Kindly check and approve the PR 

@hemantxpatel  Kindly let me know if there is any change for same.

**Review by paullatzelsperger — COMMENTED** [review]
one small issue still, see inline comment. Also, @ravinderkumarsap could you please link the most recent successful run of `verify.yaml` on the PR branch in your fork?

File: mxd/postman/MXD_Management_API.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ravinderkumarsap) [reply mxd/postman/MXD_Management_API.json:1]: @paullatzelsperger  I have committed the file **"mxd-management-apis.json"** accidentally. Now I have removed it.

[the remark this answers is not in the document]
Comment (ravinderkumarsap) [reply mxd/postman/MXD_Management_API.json:1]: @paullatzelsperger I am sharing the link for most recent successful for verify.yaml from my fork branch 
https://github.com/sap-contributions/eclipse-tractusx_tutorial-resources/actions/runs/6353886618/job/17259349437

**Review by paullatzelsperger — APPROVED** [review]
looks good

**Pull request merged**